modules/seqexec/server/src/main/scala/seqexec/server/gnirs/Gnirs.scala

```diff
@@ -89,35 +91,49 @@ object Gnirs {
     case _                 => getCCOtherConfig(config)
   }
 
+  // Used for optional keys. If the key is not present, the result is a None. Other extraction errors are passed on.
+  private implicit class KeyAttemptToOption[T](v: Either[ExtractFailure, T]) {
+    def optionalKey: Either[ExtractFailure, Option[T]] = v match {
```

Comment:
I think this could be done as well with `recoverWith`, but it is fine as it is as well

Reply:
`recoverWith` goes on the bad path and does not change the good values. I could have used `v.map(Some).recoverWith{...}`, but I think this way is fine.